src/app/beer.service.ts

```diff
@@ -0,0 +1,73 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+
+import { Observable, of } from 'rxjs';
+import { catchError, map, tap } from 'rxjs/operators';
+
+import { Beer } from './beer';
+import { MessageService } from './message.service';
+
+const httpOptions = {
+    headers: new HttpHeaders({ 'Content-Type': 'application/json' })
+};
+
+@Injectable({ providedIn: 'root' })
+export class BeerService {
+
+    private beersUrl = 'api/beers';
+
+    constructor(private http: HttpClient, private messageService: MessageService) { }
+
+    public getBeers(): Observable<Beer[]> {
+        return this.http.get<Beer[]>(this.beersUrl).pipe(
+            tap(beers => this.log(`fetched beers`)),
+            catchError(this.handleError('getBeers', []))
```

Comment:
^did you actually write this, or is this copy-pasted from some external source? :unamused: